drivers/gpio/gpio_ifx_cat1.c

```diff
@@ -0,0 +1,317 @@
+			gpio_mode = CYHAL_GPIO_DRIVE_STRONG;
+			pin_val = (flags & GPIO_OUTPUT_INIT_HIGH) ? true : false;
+		}
+		break;
+
+	case GPIO_DISCONNECTED:
+		cyhal_gpio_free(gpio_pin);
+		return 0;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	status = cyhal_gpio_init(gpio_pin, gpio_dir, gpio_mode, pin_val);
+
+	/* If the gpio requested resource is already in use, try to free and
+	 * initialize again
+	 */
+	if (status == CYHAL_HWMGR_RSLT_ERR_INUSE) {
+		cyhal_gpio_free(gpio_pin);
+		status = cyhal_gpio_init(gpio_pin, gpio_dir, gpio_mode, pin_val);
+	}
+
+	return (status == CY_RSLT_SUCCESS) ? 0 : -EIO;
+}
+
+static int gpio_cat1_port_get_raw(const struct device *dev,
+				  uint32_t *value)
+{
+	const struct gpio_cat1_config *const cfg = dev->config;
+	GPIO_PRT_Type *const base = cfg->regs;
+
+	*value = GPIO_PRT_IN(base);
+
+	return 0;
+}
+
+static int gpio_cat1_port_set_masked_raw(const struct device *dev,
+					 uint32_t mask, uint32_t value)
+{
+	const struct gpio_cat1_config *const cfg = dev->config;
+	GPIO_PRT_Type *const base = cfg->regs;
+
+	GPIO_PRT_OUT(base) = (GPIO_PRT_OUT(base) & ~mask) | (mask & value);
+
+	return 0;
+}
+
+static int gpio_cat1_port_set_bits_raw(const struct device *dev,
+				       uint32_t mask)
+{
+	const struct gpio_cat1_config *const cfg = dev->config;
+	GPIO_PRT_Type *const base = cfg->regs;
+
+	GPIO_PRT_OUT_SET(base) = mask;
+
+	return 0;
+}
+
+static int gpio_cat1_port_clear_bits_raw(const struct device *dev,
+					 uint32_t mask)
+{
+	const struct gpio_cat1_config *const cfg = dev->config;
+	GPIO_PRT_Type *const base = cfg->regs;
+
+	GPIO_PRT_OUT_CLR(base) = mask;
+
+	return 0;
+}
+
+static int gpio_cat1_port_toggle_bits(const struct device *dev,
+				      uint32_t mask)
+{
+	const struct gpio_cat1_config *const cfg = dev->config;
+	GPIO_PRT_Type *const base = cfg->regs;
+
+	GPIO_PRT_OUT_INV(base) = mask;
+
+	return 0;
+}
+
+static uint32_t gpio_cat1_get_pending_int(const struct device *dev)
+{
+	const struct gpio_cat1_config *const cfg = dev->config;
+	GPIO_PRT_Type *const base = cfg->regs;
+
+	return GPIO_PRT_INTR_MASKED(base);
+}
+
+static void gpio_event_callback(void *callback_arg, cyhal_gpio_event_t event)
+{
+	ARG_UNUSED(event);
+	uint32_t port_num = CYHAL_GET_PORT((uint32_t) callback_arg);
+	uint32_t pin_num = CYHAL_GET_PIN((uint32_t) callback_arg);
+
+	static const struct device *const port_dev_obj[IOSS_GPIO_GPIO_PORT_NR] = {
+		GET_DEV_OBJ(gpio_prt0),  GET_DEV_OBJ(gpio_prt1),  GET_DEV_OBJ(gpio_prt2),
+		GET_DEV_OBJ(gpio_prt3),  GET_DEV_OBJ(gpio_prt4),  GET_DEV_OBJ(gpio_prt5),
+		GET_DEV_OBJ(gpio_prt6),  GET_DEV_OBJ(gpio_prt7),  GET_DEV_OBJ(gpio_prt8),
+		GET_DEV_OBJ(gpio_prt9),  GET_DEV_OBJ(gpio_prt10), GET_DEV_OBJ(gpio_prt11),
+		GET_DEV_OBJ(gpio_prt12), GET_DEV_OBJ(gpio_prt13), GET_DEV_OBJ(gpio_prt14)
+	};
+
+	const struct device *dev = port_dev_obj[port_num];
+
+	/* Goes through and fires callback from a callback list */
+	gpio_fire_callbacks(&((struct gpio_cat1_data *const)(dev)->data)->callbacks,
```

Comment:
you're dereferencing dev without checking if NULL (it can be according to GET_DEV_OBJ).